### PR TITLE
[pytest][neighbor-mac] Convert Neighbor MAC no PTF test case

### PR DESCRIPTION
**Pull request by tahmed-dev**
### Description of PR
Summary:
This PR converts Ansible neighbor-mac-noptf test case to pytest framework.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Approach
#### What is the motivation for this PR?
Closing gap between Ansible and pytest framework

#### How did you do it?
Converted .yml test to Python pytest

#### How did you verify/test it?
```
tests$ pytest arp/test_neighbor_mac_noptf.py --testbed=vms12-t0-s6000-1 --inventory=../ansible/str --testbed_file=../ansible/testbed.csv --host-pattern=str-s6000-acs-14 --module-path=../ansible/library --disable_loganalyzer --skip_sanity
================================================================================================================================================== test session starts ==================================================================================================================================================
platform linux2 -- Python 2.7.12, pytest-4.6.9, py-1.8.1, pluggy-0.13.1
ansible: 2.8.7
rootdir: /var/host-acs-mgmt-repo/tests, inifile: pytest.ini
plugins: ansible-2.2.2
collected 2 items                                                                                                                                                                                                                                                                                                       

arp/test_neighbor_mac_noptf.py ..                                                                                                                                                                                                                                                                                 [100%]

============================================================================================================================================== 2 passed in 295.24 seconds ===============================================================================================================================================
```
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

File: tests/arp/test_neighbor_mac_noptf.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
+import logging
+import pytest
+import time
+
+from common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
```
Comment (wangxin) [inline tests/arp/test_neighbor_mac_noptf.py:8]: Need to add topology mark
```
pytestmark = [
    pytest.mark.topology('any')
]
```

Comment (tahmed-dev) [reply tests/arp/test_neighbor_mac_noptf.py:8]: Ack!

File: tests/arp/test_neighbor_mac_noptf.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        yield
+ 
+        logger.info("Reload Config DB")
+        duthost.shell(argv=["config", "reload", "-y"])
```
Comment (yxieca) [inline tests/arp/test_neighbor_mac_noptf.py:37]: Use https://github.com/Azure/sonic-mgmt/blob/master/tests/common/config_reload.py instead?

File: tests/arp/test_neighbor_mac_noptf.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        """
+            Verify orchagent is running before and after the test is finished
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        def getOrchagentPid(duthost):
```
Comment (yxieca) [inline tests/arp/test_neighbor_mac_noptf.py:87]: This is not a getter? it didn't return anything.

File: tests/arp/test_neighbor_mac_noptf.py (hunk 2 above)
Comment (tahmed-dev) [reply tests/arp/test_neighbor_mac_noptf.py]: Done

File: tests/arp/test_neighbor_mac_noptf.py (hunk 3 above)
Comment (tahmed-dev) [reply tests/arp/test_neighbor_mac_noptf.py:89]: good catch. Will rename. I had in mind to return the pid and have the caller asserts, then relocated the assert!

File: tests/arp/test_neighbor_mac_noptf.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            """
+                Retreive orchagent process ID
+
+                Args:
+                    duthost (AnsibleHost): Device Under Test (DUT)
+            """
+            result = duthost.shell(argv=["pgrep", "orchagent"])
+            pytest_assert(int(result["stdout"]) > 0, "Orchagent is not running")
```
[the remark this answers is not in the document]
Comment (tahmed-dev) [reply tests/arp/test_neighbor_mac_noptf.py:97]: This is the output of pgrep, it should be the pid of orchagent

**Pull request merged**